Source/astcenc_vecmathlib_neon_armv7_4.h

```diff
@@ -0,0 +1,174 @@
+// under the License.
+// ----------------------------------------------------------------------------
+
+/**
+ * @brief Intrinsics for Armv7 NEON.
+ *
+ * This module implements a few Armv7-compatible intrinsics indentical to Armv8
+ * ones. Thus, asticenc can be compiled using Armv7 architecture.
+ */
+
+#ifndef ASTC_VECMATHLIB_NEON_ARMV7_4_H_INCLUDED
+#define ASTC_VECMATHLIB_NEON_ARMV7_4_H_INCLUDED
+
+#ifndef ASTCENC_SIMD_INLINE
+	#error "Include astcenc_vecmathlib.h, do not include directly"
+#endif
+
+#include <algorithm>
+
+/**
+ * @brief Return the max vector of two vectors.
+ *
+ * If one vector element is numeric and the other is a quiet NaN,
+ * the result placed in the vector is the numerical value.
+ */
+ASTCENC_SIMD_INLINE float32x4_t vmaxnmq_f32(float32x4_t a, float32x4_t b)
+{
+	uint32x4_t amask = vceqq_f32(a, a);
+	uint32x4_t bmask = vceqq_f32(b, b);
+	a = vbslq_f32(amask, a, b);
+	b = vbslq_f32(bmask, b, a);
+	return vmaxq_f32(a, b);
+}
+
+/**
+ * @brief Return the min vector of two vectors.
+ *
+ * If one vector element is numeric and the other is a quiet NaN,
+ * the result placed in the vector is the numerical value.
+ */
+ASTCENC_SIMD_INLINE float32x4_t vminnmq_f32(float32x4_t a, float32x4_t b)
+{
+	uint32x4_t amask = vceqq_f32(a, a);
+	uint32x4_t bmask = vceqq_f32(b, b);
+	a = vbslq_f32(amask, a, b);
+	b = vbslq_f32(bmask, b, a);
+	return vminq_f32(a, b);
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE float32_t vmaxvq_f32(float32x4_t a)
+{
+	float32_t a0 = vgetq_lane_f32(a, 0);
+	float32_t a1 = vgetq_lane_f32(a, 1);
+	float32_t a2 = vgetq_lane_f32(a, 2);
+	float32_t a3 = vgetq_lane_f32(a, 3);
+	return std::max(std::max(a0, a1), std::max(a2, a3));
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE float32_t vminvq_f32(float32x4_t a)
+{
+	float32_t a0 = vgetq_lane_f32(a, 0);
+	float32_t a1 = vgetq_lane_f32(a, 1);
+	float32_t a2 = vgetq_lane_f32(a, 2);
+	float32_t a3 = vgetq_lane_f32(a, 3);
+	return std::min(std::min(a0, a1), std::min(a2, a3));
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE int32_t vmaxvq_s32(int32x4_t a)
+{
+	int32_t a0 = vgetq_lane_s32(a, 0);
+	int32_t a1 = vgetq_lane_s32(a, 1);
+	int32_t a2 = vgetq_lane_s32(a, 2);
+	int32_t a3 = vgetq_lane_s32(a, 3);
+	return std::max(std::max(a0, a1), std::max(a2, a3));
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE int32_t vminvq_s32(int32x4_t a)
+{
+	int32_t a0 = vgetq_lane_s32(a, 0);
+	int32_t a1 = vgetq_lane_s32(a, 1);
+	int32_t a2 = vgetq_lane_s32(a, 2);
+	int32_t a3 = vgetq_lane_s32(a, 3);
+	return std::min(std::min(a0, a1), std::min(a2, a3));
+}
+
+/**
+ * @brief Return the sqrt of the lanes in the vector.
+ */
+ASTCENC_SIMD_INLINE float32x4_t vsqrtq_f32(float32x4_t a)
+{
+	a = vmaxq_f32(a, vdupq_n_f32((std::numeric_limits<float>::min)()));
```

Comment:
Please replace with a scalarized version using `std::sqrt()`. Reciprocal sqrt is not invariant across implementations, and we have a strict invariance requirement.